ml/src/test/scala/frameless/ml/regression/TypedRandomForestRegressorTests.scala

```diff
@@ -0,0 +1,74 @@
+package frameless
+package ml
+package regression
+
+import shapeless.test.illTyped
+import org.apache.spark.ml.linalg._
+import frameless.ml.regression.TypedRandomForestRegressor.FeatureSubsetStrategy
+import org.scalacheck.Arbitrary
+import org.scalacheck.Prop._
+import org.scalatest.MustMatchers
+
+class TypedRandomForestRegressorTests extends FramelessMlSuite with MustMatchers {
+  implicit val arbVectorNonEmpty: Arbitrary[Vector] =
+    Arbitrary(Generators.arbVector.arbitrary suchThat (_.size > 0)) // vector must not be empty for RandomForestRegressor
+
+  test("fit() returns a correct TypedTransformer") {
+    val prop = forAll { x2: X2[Double, Vector] =>
+      val rf = TypedRandomForestRegressor.create[X2[Double, Vector]]()
+      val ds = TypedDataset.create(Seq(x2))
+      val model = rf.fit(ds)
+      val pDs = model.transform(ds).as[X3[Double, Vector, Double]]
+
+      pDs.select(pDs.col('a), pDs.col('b)).collect.run() == Seq(x2.a -> x2.b)
+    }
+
+    val prop2 = forAll { x2: X2[Vector, Double] =>
+      val rf = TypedRandomForestRegressor.create[X2[Vector, Double]]()
+      val ds = TypedDataset.create(Seq(x2))
+      val model = rf.fit(ds)
+      val pDs = model.transform(ds).as[X3[Vector, Double, Double]]
+
+      pDs.select(pDs.col('a), pDs.col('b)).collect.run() == Seq(x2.a -> x2.b)
+    }
+
+    def prop3[A: TypedEncoder: Arbitrary] = forAll { x3: X3[Vector, Double, A] =>
+      val rf = TypedRandomForestRegressor.create[X2[Vector, Double]]()
+      val ds = TypedDataset.create(Seq(x3))
+      val model = rf.fit(ds)
+      val pDs = model.transform(ds).as[X4[Vector, Double, A, Double]]
+
+      pDs.select(pDs.col('a), pDs.col('b), pDs.col('c)).collect.run() == Seq((x3.a, x3.b, x3.c))
+    }
+
+    check(prop)
+    check(prop2)
+    check(prop3[String])
+    check(prop3[Double])
+  }
+
+  test("param setting is retained") {
+    val rf = TypedRandomForestRegressor.create[X2[Double, Vector]]()
+      .setNumTrees(10)
+      .setMaxBins(100)
+      .setFeatureSubsetStrategy(FeatureSubsetStrategy.All)
```

Comment:
It looks like you only tested `All`, could you add something for the other cases?

Reply:
@OlivierBlanvillain BTW, for params like `numTrees`, `maxDepth` etc that have a bounded valid range, I was thinking about using https://github.com/fthomas/refined to check at compile the correctness of the parameter. Is including refined in frameless-ml build seems ok for you?

Reply:
I would stick to shapeless for now, you could use `NatProductArgs` and ask for a `nat.obs.Range[_0, _100]`. But don't that on this PR, it's already getting pretty big :)

Reply:
Ok :)